cinder/volume/drivers/nexenta/nfs.py

```diff
@@ -1006,15 +1000,23 @@ def _update_volume_stats(self):
             free_capacity_gb = 'unknown'
             total_capacity_gb = 'unknown'
         else:
-            free_capacity_gb = stats['available'] // units.Gi
-            allocated_capacity_gb = stats['used'] // units.Gi
+            free_capacity_gb = stats['available']
+            allocated_capacity_gb = stats['used']
             total_capacity_gb = free_capacity_gb + allocated_capacity_gb
         provisioned_capacity_gb = total_volumes = 0
         ctxt = context.get_admin_context()
         volumes = objects.VolumeList.get_all_by_host(ctxt, self.host)
         for volume in volumes:
             provisioned_capacity_gb += volume['size']
             total_volumes += 1
+        if self.volume_compression:
```

Comment:
same here

Reply:
self.volume_compression is self.configuration.nexenta_dataset_compression. and self.configuration.nexenta_dataset_compression has no default value.